src/modules/verification-service/verification.service.ts

```diff
@@ -87,7 +89,9 @@ export class VerificationService implements IVerificationService {
     };
 
     const provider: INinVerificationProvider =
-      await this.verificationProviderFactory.makeVerificationProvider(input);
+      (await this.verificationProviderFactory.makeVerificationProvider(
+        input,
+      )) as YouVerifyProvider;
```

Comment:
Same here, concrete implementation violating DIP. Now If I decide for example to swap out YouVerify for Trulioo or Paystack, I will have to change all the service code that are depending directly on it e.g this.
Whereas with my implementation I can manipulate the specific provider implementation without touching any of the actual business logic